### PR TITLE
Feature/this error

### PR DESCRIPTION
**Pull request by Abraxas-365**

### CODE REVIEW

File: src/text_splitter/error.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    EmptyInputText,
+
+    #[error("Mismatch metadatas and text")]
+    MetaDataTextMismatch,
+
+    #[error("Tokenizer not found")]
+    TokenizerNotFound,
+
+    #[error("Tokenizer creation failed: {0}")]
+    TokenizerCreationFailed(#[from] anyhow::Error),
```
Comment (prabirshrestha) [inline src/text_splitter/error.rs:15]: Do we really need `anyhow::Error` usually should be avoided it in libraries if possible.

File: src/text_splitter/error.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TextSplitterError {
+    #[error("Empty input text")]
+    EmptyInputText,
+
+    #[error("Mismatch metadatas and text")]
+    MetaDataTextMismatch,
```
Comment (prabirshrestha) [inline src/text_splitter/error.rs:9]: we should fix the casing to be `MetadataTextMismatch` with small `d`.

File: src/text_splitter/error.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TextSplitterError {
+    #[error("Empty input text")]
+    EmptyInputText,
+
+    #[error("Mismatch metadatas and text")]
```
Comment (prabirshrestha) [inline src/text_splitter/error.rs:8]: should not have `s` in `metadata`.

**Review by prabirshrestha — APPROVED** [review]
LGTM. Add few nit comments.

File: src/text_splitter/error.rs (hunk 1 above)
Comment (Abraxas-365) [reply src/text_splitter/error.rs:15]: One of the libraries we use, its using it , so im creating a from any how

File: src/text_splitter/error.rs (hunk 2 above)
Comment (Abraxas-365) [reply src/text_splitter/error.rs:9]: fixing it 😄

File: src/text_splitter/error.rs (hunk 1 above)
Comment (Abraxas-365) [reply src/text_splitter/error.rs]: Im creating an error and deliting the use of anyhow

**Pull request merged**